contracts/Market.sol

```diff
@@ -791,7 +785,7 @@ contract Market is
     uint8 v,
     bytes32 r,
     bytes32 s
-  ) external whenNotPaused {
+  ) external whenNotPaused onlyRole(MARKET_ADMIN_ROLE) {
```

Comment:
This is the only change

Reply:
what's the purpose of allowlist_role when we permission based on this too?

Reply:
Allowlist will still allow you to buy from the restricted threshold pool.
If we decide to keep this functionality and allow buyers direct access to it, then obviously we'd use a role other than MARKET_ADMIN_ROLE. But I also think that role should be distinct from the ALLOWLIST_ROLE.

Reply:
Relatedly, we could consider removing `_validatePrioritySupply` from `_allocateRemovalsFromSupplier` since it's now safe to assume we only ever allocate removals from a specific supplier when the caller already has the MARKET_ADMIN role. Would reduce gas cost. That's also assuming we don't ever want to enforce the `ALLOWLIST_ROLE` check on someone who has the `MARKET_ADMIN_ROLE`, which seems unnecessary.

But again, at some point in the future I anticipate introducing a third role for allowing buyers to buy from specific suppliers, and then we'd have to put this check back in if we removed it here and now.

Reply:
RE roles: I think it's useful to have more granular roles so we can, for example, allow third party wallet/contract interpretations.  

RE _validatePrioritySupply: if it makes sense to remove that, that would be great. Not sure if my prior point negates the feasibility of that or not. I do find it odd/suboptimal that we check supply + priority supply in different ways across different functions (e.g., `_allocateRemovals` vs `_allocateSupplySingleSupplier`)

Reply:
[Here's a draft PR ](https://github.com/nori-dot-eco/contracts/pull/680)with the changes needed.
It has the added benefit of not making me to fix the tests so that we have an admin account who does not also have the `ALLOWLIST_ROLE` (if I'm correctly understanding why that test is failing)!

Reply:
There's also a second `swapFromSupplier` you need to permission, I believe.

Reply:
Add this to the list of requirements in the natspec
 ```
- Can only be used when the caller has the `MARKET_ADMIN_ROLE` role.
```